src/test/java/org/killbill/billing/plugin/adyen/api/TestAdyenPaymentPluginApi.java

```diff
@@ -414,6 +439,113 @@ public void testHPP() throws Exception {
         System.out.flush();
     }
 
+    @Test(groups = "slow")
+    public void testAuthorizeAndComplete3DSecure() throws Exception {
+        adyenPaymentPluginApi.addPaymentMethod(payment.getAccountId(), payment.getPaymentMethodId(), adyenPaymentMethodPluginCC(), true, propertiesWith3DSInfo, context);
+        final PaymentTransaction authorizationTransaction1 = TestUtils.buildPaymentTransaction(payment, TransactionType.AUTHORIZE, DEFAULT_CURRENCY);
+        final PaymentTransaction authorizationTransaction2 = TestUtils.buildPaymentTransaction(payment, TransactionType.AUTHORIZE, DEFAULT_CURRENCY);
+        final PaymentTransaction captureTransaction = TestUtils.buildPaymentTransaction(payment, TransactionType.CAPTURE, DEFAULT_CURRENCY);
+
+        final PaymentTransactionInfoPlugin authorizationInfoPlugin1 = adyenPaymentPluginApi.authorizePayment(payment.getAccountId(),
+                                                                                                             payment.getId(),
+                                                                                                             authorizationTransaction1.getId(),
+                                                                                                             payment.getPaymentMethodId(),
+                                                                                                             authorizationTransaction1.getAmount(),
+                                                                                                             authorizationTransaction1.getCurrency(),
+                                                                                                             propertiesWith3DSInfo,
+                                                                                                             context);
+
+        assertEquals(authorizationInfoPlugin1.getGatewayError(), "RedirectShopper");
```

Comment:
For historical reasons, there is only one field exposed at the API level ([gatewayError](https://github.com/killbill/killbill-api/blob/master/src/main/java/org/killbill/billing/payment/plugin/api/PaymentTransactionInfoPlugin.java#L75)), so we overload it with generic status messages in the happy path.

That being said, there is some work to fine-tune what's populated (tracked by https://github.com/killbill/killbill-adyen-plugin/issues/15).